cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java

```diff
@@ -0,0 +1,529 @@
+        when(mockRebalanceOps.getAsync(CLUSTER_NAMESPACE, RESOURCE_NAME)).thenReturn(Future.succeededFuture(kcRebalance));
+
+        return kcrao.computeNextStatus(
+                recon, HOST, client, kcRebalance, currentState, initialAnnotation, rbOptions).compose(result -> {
+                    context.verify(() -> {
+                        assertTrue(expectedStatusCheck(result, nextState));
+                    });
+                    return Future.succeededFuture(result);
+                });
+    }
+
+    /**
+     *  Checks the expected transition between two states of the Kafka Rebalance operator.
+     *
+     * @param vertx The vertx test instance.
+     * @param context The test context instance.
+     * @param currentState The current state of the resource before being passed to computeNextStatus.
+     * @param nextState The expected state of the resouce after computeNextStatus has been called.
+     * @param initialAnnotation The initial annotation attached to the Kafka Rebalance resource. For example none or refresh.
+     * @param userAnnotation An annotation to be applied after the reconcile has started, for example "approve" or "stop".
+     * @param currentStatusSessionID The user task ID attached to the current KafkaRebalance resource. Can be null.
+     * @return A future for the {@link KafkaRebalanceStatus} returned by the {@link KafkaRebalanceAssemblyOperator#computeNextStatus} method
+     */
+    private Future<KafkaRebalanceStatus> checkTransition(Vertx vertx, VertxTestContext context,
+                                                         KafkaRebalanceAssemblyOperator.State currentState,
+                                                         KafkaRebalanceAssemblyOperator.State nextState,
+                                                         KafkaRebalanceAssemblyOperator.RebalanceAnnotation initialAnnotation,
+                                                         String userAnnotation, String currentStatusSessionID) {
+
+        KafkaRebalance kcRebalance = createKafkaRebalance(currentState, currentStatusSessionID, userAnnotation, null);
+
+        return checkTransition(vertx, context, currentState, nextState, initialAnnotation, kcRebalance);
+
+    }
+
+    private static void defaultStatusHandler(AsyncResult<KafkaRebalanceStatus> result, VertxTestContext context) {
+        if (result.succeeded()) {
+            context.completeNow();
+        } else {
+            context.failNow(result.cause());
+        }
+    }
+
+    private static void checkOptimizationResults(AsyncResult<KafkaRebalanceStatus> result, VertxTestContext context, boolean shouldBeEmpty) {
+        if (result.succeeded()) {
+            if (shouldBeEmpty) {
+                context.verify(() -> assertTrue(result.result().getOptimizationResult().isEmpty()));
+            } else {
+                context.verify(() -> assertFalse(result.result().getOptimizationResult().isEmpty()));
+            }
```

Comment:
Why use an if when you could write `assertEquals(shouldBeEmpty, result.result().getOptimizationResult().isEmpty())` ?

Reply:
Why not indeed?